snapshot/pkg/controller/snapshotter/snapshotter.go

```diff
@@ -286,7 +291,11 @@ func (vs *volumeSnapshotter) deleteSnapshot(spec *crdv1.VolumeSnapshotDataSpec)
 		return fmt.Errorf("%s is not supported volume for %#v", volumeType, spec)
 	}
 	source := spec.VolumeSnapshotDataSource
-	err := plugin.SnapshotDelete(&source, nil /* *v1.PersistentVolume */)
+	pv, err := vs.getPVFromName(spec.PersistentVolumeRef.Name)
+	if err != nil {
+		glog.Warningf("failed to retrieve PV %s from the API server: %q", spec.PersistentVolumeRef.Name, err)
```

Comment:
I'm not quite sure about the warning here: none of the plugins actually require a PV to delete the snapshot. I think it's a good idea to try to get the PV since we have it in the interface (just in the case some plugin needs it in the future). I would leave it up to the plugin to complain about the missing PV eventually.

Reply:
Logic of snapshot should be independent of plugins. That is why I think controller should log its own warnings. Further, it is possible that plugin does not make proper check for nil values. In this case this log would be helpful.

Reply:
If you are not convinced then I can remove this.

Reply:
Hm... It's not that important and you're right. The API was written to be future-proof but with the CSI on the way we're unlikely to see new plugins added here. So your change is OK for the current state of things. Maybe we should simplify the API and don't require the PV for snapshot deletion -- it would save us some API calls too.